pandas/tests/indexing/test_partial.py

```diff
@@ -469,22 +469,21 @@ def f():
         expected["foo"] = expected["foo"].astype("float64")
 
         def f():
-            df = DataFrame()
+            df = DataFrame(index=Index([], dtype="int64"))
             df["foo"] = []
             return df
 
         tm.assert_frame_equal(f(), expected)
 
         def f():
```

Comment:
why do these need nested functions?

Reply:
Agreed it's strange, but we'll ignore that, since the modifications are following the existing structure.